### PR TITLE
[SPARK-49504][BUILD] Add `jjwt` profile

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to add a new profile `jjwt` to provide `jjwt-impl` and `jjwt-jackson` jars files in a Spark distribution.

### Why are the changes needed?

To provide an easy way to build a Spark distribution.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually.

**BEFORE**
```
$ mvn dependency:tree --pl assembly --am | grep jjwt-impl
[INFO] +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
```

**AFTER**
```
$ mvn dependency:tree --pl assembly --am -Pjjwt | grep jjwt-impl
[INFO] +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:compile
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:test
[INFO] |  +- io.jsonwebtoken:jjwt-impl:jar:0.12.6:compile
```

Or, build the distribution. 

```
$ dev/make-distribution.sh -Pjjwt
$ ls dist/jars/jj*
dist/jars/jjwt-api-0.12.6.jar
dist/jars/jjwt-impl-0.12.6.jar
dist/jars/jjwt-jackson-0.12.6.jar
```
### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -626,6 +621,12 @@
         <script.extension>.sh</script.extension>
       </properties>
     </profile>
+    <profile>
+      <id>jjwt</id>
+      <properties>
+        <jjwt.deps.scope>compile</jjwt.deps.scope>
```
Comment (dongjoon-hyun) [inline core/pom.xml:627]: We need this both here and `assembly` in order to include the dependencies properly.

**Comment by dongjoon-hyun** [comment]
Could you review this when you have some time, @viirya ?

**Review by HyukjinKwon — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @viirya !

**Pull request closed without merging**